staging/src/k8s.io/client-go/tools/cache/listwatch.go

```diff
@@ -87,6 +91,9 @@ func timeoutFromListOptions(options metav1.ListOptions) time.Duration {
 
 // List a set of apiserver resources
 func (lw *ListWatch) List(options metav1.ListOptions) (runtime.Object, error) {
+	if !lw.DisablePaging {
```

Comment:
@smarterclayton - did you do any performance-related experiments?

E.g. if I want to list 50k of objects, what is the different with pagination for e2e time of getting the result?